hivemind/server/connection_handler.py

```diff
@@ -60,11 +60,13 @@ async def info(self, request: runtime_pb2.ExpertUID, context: grpc.ServicerConte
     async def forward(self, request: runtime_pb2.ExpertRequest, context: grpc.ServicerContext):
         inputs = [deserialize_torch_tensor(tensor) for tensor in request.tensors]
         future = self.experts[request.uid].forward_pool.submit_task(*inputs)
-        serialized_response = [serialize_torch_tensor(tensor) for tensor in await future]
+        serialized_response = [serialize_torch_tensor(tensor, proto.compression) for tensor, proto in zip(
+            await future, nested_flatten(self.experts[request.uid].outputs_schema))]
```

Comment:
```suggestion
        serialized_response = [serialize_torch_tensor(tensor, proto.compression, allow_inplace=True) for tensor, proto
                               in zip(await future, nested_flatten(self.experts[request.uid].outputs_schema))]

```

rationale: added allow_inplace=True